Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -513,14 +513,24 @@
 
     // Give array of advances of zero so it will use positions correctly
     std::vector<FLOAT> dwriteAdvances(count, 0);
-
     DWRITE_GLYPH_RUN run = { fontFace.Get(), curState->fontSize, count, glyphs, dwriteAdvances.data(), positions.data(), FALSE, 0 };
+
+    // CGContextShow* functions transform the glyphs about the origin of the first glyph, rather than about each glyph's origin
```

Comment:
Both, but mostly to explain why we apply the text transform to ctm and pretend text matrix is identity